### DISCUSSION TITLE
在iPhone上的浏览器，PAGVIEW初始化有概率会卡住，后续的所有的pag就无法加载

### DISCUSSION THREAD

**Discussion by yc904573814** [Tencent/libpag#2063] [Q&A]
vue2 环境，sdk版本4.3.45，
ios端
在初始化完PAGInit方法后 拿到的PAG去初始化PAGView，这个过程有概率会卡住，就不会进行下去了。
测试多台iPhone，有一台会在卡住的情况下有概率会报错，如图
![微信图片_20240105174608](https://github.com/Tencent/libpag/assets/57278903/e1efcdc2-320e-4e03-830a-0901aabb862f)
就一直await 中 进行不下去了，当这个问题出现后，别的pagView也无法加载，除非重新刷新页面
安卓机没问题
```
that.pagView = await PAG.PAGView.init(res.pagFile, "#pag");
```
pag文件
[test.zip](https://github.com/Tencent/libpag/files/13840494/test.zip)

**Comment by DaYang816** [comment]
我也有这个问题，特别当iPhone在微信浏览器中，重现概率特别高，PAGView.init直接呆住下不去

**Comment by newlife70723** [comment]
我再WEB也有類似問題，在播放次數過較高時，會出現動畫停住的狀況

**Comment by Comori** [comment]
我也遇到了类似的问题，看了下wasm代码，应该是下面报错，但是不知道为什么。
```typescript
function instantiateAsync() {
    if (!wasmBinary && typeof WebAssembly.instantiateStreaming == "function" && !isDataURI(wasmBinaryFile) && typeof fetch == "function") {
        return fetch(wasmBinaryFile, { credentials: "same-origin" })
            .then(function (response) {
                var result = WebAssembly.instantiateStreaming(response, info);
                return result.then(receiveInstantiationResult, function (reason) {
                    err("wasm streaming compile failed: " + reason); err("falling back to ArrayBuffer instantiation");
                    return instantiateArrayBuffer(receiveInstantiationResult)
                })
            })
    } else {
        return instantiateArrayBuffer(receiveInstantiationResult)
    }
}
```

**Comment by 95sunny** [comment]
我也经常卡住， web端， 经常会出现一个大概是WebAssembly内存不足的异常，导致页面整体严重卡顿，像是内存泄漏了。
复现概率不大， 复现方式就是开发过程长时间挂着，多次刷新加载。 
猜测是不是强刷操作导致没有正确回收WebAssembly相关的内存，导致了泄漏

**Comment by CodeJhF** [comment]
感谢 @95sunny 的回复。

因长期无回复，且 libpag 4.3 及更早版本已停止维护（当前仅维护 4.4 / 4.5），先关闭本条。
上报版本：4.3.45。

如果问题在最新版本（4.4 / 4.5）仍能复现，欢迎在本帖回复或新开 issue，并附上 libpag 版本号、平台与系统版本、可复现的 PAG 文件或最小 demo，我们会优先跟进。（stale-close）
